### PR TITLE
Attach warning to docs of functions in experimental

### LINKED ISSUE #2011: Make the user aware of that they are using code from `experimental`
I decided to merge #1999 and relocate the discussion on the separation of `experimental` from the rest of the source here.
- We all seemed to agree that the user should in some way be made aware of that they are using code from `experimental`. Just for the sake of completeness: Please disagree here if you disagree.
- Should code from experimental be available by default? As already stated, if not its discoverability goes down, but that is not necessarily a bad thing. In my pov code should not stay "experimental" for too long.

These were the ideas of @fingolfin :
> Indeed. Some options come to mind:
> 
>     * maybe accessing code in experimental should be "guarded" and require `using Oscar.Experimental`, or require writing `Experimantal.FOO` (so `Experimental.galois_group(...)`
>       
>       * pro: obviously experimental
>       * con: less discoverable
> 
>     * or maybe the primary functions should have a `@warn_experimental()` call in them, which prints a warning about them being experimental (possibly only the first time they are called)
> 
>     * ... other ideas?

So I'll cc some people who might be interested: @micjoswig @fieker @benlorenz . Maybe we can discuss a bit here and then decide during one of the next Friday meetings, preferably when we are all there, unless we immediately find a consensus here.

### PR DESCRIPTION
**Pull request by lkastner**
preview: https://docs.oscar-system.org/previews/PR3709/

### CODE REVIEW

File: experimental/Experimental.jl [hunk 1]
```diff
@@ -57,3 +57,29 @@ for pkg in oldexppkgs
   # Load the package
   include(joinpath(expdir, pkg, "$pkg.jl"))
 end
+
+# We modify the documentation of the experimental part to attach a warning to
+# every exported function that this function is part of experimental.
+# Furthermore we give a link for users to rea up on what this entails.
```
Comment (fingolfin) [inline experimental/Experimental.jl]:
```suggestion
# Furthermore we give a link for users to read up on what this entails.
```

**Review by fingolfin — COMMENTED** [review]
Nice!

Do we need a specific Documenter version for this to work? Then perhaps specify that in docs/Project.toml?

File: experimental/Experimental.jl [hunk 2]
```diff
@@ -57,3 +57,29 @@ for pkg in oldexppkgs
...
+#
+# Note that there are functions in the docs of experimental that are not
+# exported. These then also do not get the warning attached.
+using Markdown
+warnexp = Markdown.parse(raw"""
+!!! warning "Experimental"
+    This function is part of the experimental code in Oscar. Please read
+    [here](https://docs.oscar-system.org/dev/Experimental/intro/) for more
```
Comment (lgoettgens) [inline experimental/Experimental.jl]:
```suggestion
    [here](https://docs.oscar-system.org/v1/Experimental/intro/) for more
```
Or maybe make this dependent on `Oscar.is_dev`?

**Review by lgoettgens — COMMENTED** [review]
Nice addition!! One small comment above

File: Project.toml [hunk 3]
```diff
@@ -12,6 +12,7 @@ Hecke = "3e1990a7-5d81-5526-99ce-9ba3ff248f21"
 JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
 JSON3 = "0f8b85d8-7281-11e9-16c2-39a750bddbf1"
 LazyArtifacts = "4af54fe1-eca0-43a8-85a7-787d91b784e3"
+Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
```
Comment (lgoettgens) [inline Project.toml:15]: this needs a compat of something like `"1.6"` (as it is a stdlib)

**Comment by lkastner** [comment]
> Do we need a specific Documenter version for this to work? Then perhaps specify that in docs/Project.toml?

It is independent of Documenter. This is only using the Julia documentation features. And not in a good way...

**Review by fingolfin — APPROVED** [review]
Looks good to me

**Comment by thofma** [comment]
Awesome, thanks @lkastner!

**Review by thofma — APPROVED** [review]

**Pull request merged**